datafusion/core/tests/sqllogictests/src/setup.rs

```diff
@@ -29,8 +29,41 @@ use datafusion::{
     test_util,
 };
 use std::sync::Arc;
+use datafusion::prelude::AvroReadOptions;
 
-use crate::utils;
+use crate::{TestContext, utils};
+
+pub async fn register_avro_tables(ctx: &mut TestContext) {
+    register_avro_test_data(ctx).await;
+}
+
+async fn register_avro_test_data(ctx: &mut TestContext) {
+    ctx.enable_testdir();
```

Comment:
👍  looks good to me -- I think it may even make sense to always create the temp directory but for now making it just for avro looks great.